src/Microsoft.AspNetCore.Blazor/Components/BindMethods.cs

```diff
@@ -23,6 +23,24 @@ public static class BindMethods
             value == default ? null
             : (format == null ? value.ToString() : value.ToString(format));
 
+        /// <summary>
+        /// Not intended to be used directly.
+        /// </summary>
+        public static string GetEventHandlerValue<T>(string value)
+            where T : UIEventArgs
+        {
+            return value;
+        }
+
+        /// <summary>
+        /// Not intended to be used directly.
+        /// </summary>
+        public static UIEventHandler GetEventHandlerValue<T>(Action<T> value)
+            where T : UIEventArgs
+        {
+            return e => value((T)e);
```

Comment:
Do you have concerns about perf with this?

If I understand correctly, this is going to create a new delegate each time, which means that any usage of `onclick` with an event handler will need to edit the dom, and will register/unregister and event handler.

I think we could avoid this if we could avoid the need to create a `UIEventHandler` instance that wraps `Action<UIMouseEventArgs>`.

Do you see any good alternatives?

Reply:
Well, I *am* concerned that we want to minimise both the GC churn and register/unregister costs of event handlers, however I can't guess whether (a) what you're doing here definitely makes it worse vs what we had before, or (b) whether there's a simple good alternative.

Perhaps we can go ahead with what you have here right now, and then when it's in I'd like to have a look in `ildasm` at what's happening as a result. If it turns out that we can avoid allocations either by caching the delegates in some way or by not having the `UIEventHandler` type at all, that would be a good subsequent enhancement.

Reply:
After offline discussion we're going to try a slightly different approach with this. Will follow up in a separate PR.